Alarm/Scenes/Timer/TimerViewController.swift

```diff
@@ -22,18 +23,56 @@ final class TimerViewController: UIViewController {
 
   private let disposeBag = DisposeBag()
   private var timerItems: [TimerItem] = []
-  
+
   override func viewDidLoad() {
     super.viewDidLoad()
-    TimerDataManager.shared.loadTimers() // 타이머 데이터 조회
-    
-    TimerDataManager.shared.timers // timers에 변동이 생기면
-      .subscribe(onNext: { [weak self] items in
-        self?.timerItems = items
-        self?.timerTableView.reloadData() // 테이블뷰 새로고침
+
+    requestNotificationPermission()  // 알림 권한 요청
+    UNUserNotificationCenter.current().delegate = self
+    TimerDataManager.shared.loadTimers()  // 타이머 데이터 조회
+
+    TimerDataManager.shared.timers  // timers 구독
+      .subscribe(onNext: { [weak self] items in  // timers에 새로운 데이터가 들어오면 호출
+        self?.timerItems = items  // 최신 데이터를 VC 상태에 반영
+        self?.timerTableView.reloadData()  // 테이블뷰 새로고침
       })
       .disposed(by: disposeBag)
-    
+
+    // 타이머(1초마다 이벤트 발생)
+    Observable<Int>.interval(.seconds(1), scheduler: MainScheduler.instance)
+      .subscribe(onNext: { _ in
+        // 현재 저장된 데이터 조회
+        guard var items = try? TimerDataManager.shared.timers.value() else { return }
+        var changed = false  // 값 변화 추적
+        var finished: [TimerItem] = []  // 0이 된 타이머들
+
+        for i in items.indices {  // 데이터 중
+          if items[i].isActive, items[i].time > 0 {  // 타이머 진행 중 + 시간이 0보다 크면
+            items[i].time -= 1  // 시간 1 감소
+            changed = true  // 값 변화 확인
+
+            if items[i].time == 0 {
+              finished.append(items[i])
+            }
+          }
+        }
+
+        finished.forEach {
+          self.notifyTimerFinished($0)
+        }
+
+        if !finished.isEmpty {
+          let finishiedIds = Set(finished.map { $0.id })
+          items.removeAll { finishiedIds.contains($0.id) }
```

Comment:
이 부분 오타난거 같은데 확인 부탁드립니다.
```suggestion
          let finishedIds = Set(finished.map { $0.id })
          items.removeAll { finishedIds.contains($0.id) }
```